Extension/src/cppTools.ts

```diff
@@ -7,46 +7,106 @@
 import { CustomConfigurationProvider, Version } from 'vscode-cpptools';
 import { CppToolsTestApi, CppToolsTestHook } from 'vscode-cpptools/out/testApi';
 import { CustomConfigurationProvider1, getCustomConfigProviders, CustomConfigurationProviderCollection } from './LanguageServer/customProviders';
+import { getOutputChannel } from './logger';
 import * as LanguageServer from './LanguageServer/extension';
 import * as test from './testHook';
 
 export class CppTools implements CppToolsTestApi {
     private version: Version;
     private providers: CustomConfigurationProvider1[] = [];
+    private timers = new Map<string, NodeJS.Timer>();
 
     constructor(version: Version) {
+        if (version > Version.latest) {
+            console.warn(`Version ${version} is not supported by this version of cpptools. Using ${Version.latest} instead.`);
+            version = Version.latest;
+        }
         this.version = version;
     }
 
-    registerCustomConfigurationProvider(provider: CustomConfigurationProvider): void {
+    private addNotifyReadyTimer(provider: CustomConfigurationProvider1): void {
+        if (this.version >= Version.v2) {
+            const timeout: number = 30;
+            let timer: NodeJS.Timer = setTimeout(() => {
+                console.warn(`registered provider ${provider.extensionId} did not call 'notifyReady' within ${timeout} seconds`);
+            }, timeout * 1000);
```

Comment:
The capitalization at the start and the period at the end is not consistent with the console.warn above. Not sure if it matters.

Reply:
There were two sentences in the first one.  That's why it was doing sentence casing.  I split it into two separate messages instead.